src/main/java/com/gmail/shablinski93/repository/DishRepository.java

```diff
@@ -4,19 +4,20 @@
 import com.gmail.shablinski93.model.Ingredient;
 
 import java.sql.Connection;
+import java.sql.SQLException;
 import java.util.List;
 import java.util.UUID;
 
 public interface DishRepository {
-    Dish addDish(Connection connection, Dish dish);
+    Dish addDish(Connection connection, Dish dish) throws SQLException;
```

Comment:
There was a suggestion to avoid catching each exception. Right now you expose exceptions outside of your API. I believe that's not what need to do here.
It's better to handle exception at.single point.
It means that you will have a service that will interact with the DB and handle all exceptions. Using this approach you can handle exceptions similar way across your applications.

It will looks like (pseudo code)

```
DBInteractionService {
  Result executeQuery(Query q) {
    try {
      .....
    } catch (SqlException e) {
      .....
    }
  }
}

ExactServiceThatRequireSQL {
     private DBInteractionService dbSrv = new DBInteractionService();
     
     List<Obj> GetEntity() {
       .....
       result = dbSrv.executeQuery(q);
       return result;
     }
}
```

Reply:
Thanks for idea!